src/main/java/com/gregtechceu/gtceu/data/tag/GTDataComponents.java

```diff
@@ -75,12 +73,11 @@ public class GTDataComponents {
                     builder -> builder.persistent(Codec.BYTE).networkSynchronized(ByteBufCodecs.BYTE));
     public static final DeferredHolder<DataComponentType<?>, DataComponentType<SimpleFluidFilter>> SIMPLE_FLUID_FILTER = DATA_COMPONENTS
             .registerComponentType("simple_fluid_filter", builder -> builder.persistent(SimpleFluidFilter.CODEC));
-    public static final DeferredHolder<DataComponentType<?>, DataComponentType<TagFluidFilter>> TAG_FLUID_FILTER = DATA_COMPONENTS
-            .registerComponentType("tag_fluid_filter", builder -> builder.persistent(TagFluidFilter.CODEC));
     public static final DeferredHolder<DataComponentType<?>, DataComponentType<SimpleItemFilter>> SIMPLE_ITEM_FILTER = DATA_COMPONENTS
             .registerComponentType("simple_item_filter", builder -> builder.persistent(SimpleItemFilter.CODEC));
-    public static final DeferredHolder<DataComponentType<?>, DataComponentType<TagItemFilter>> TAG_ITEM_FILTER = DATA_COMPONENTS
-            .registerComponentType("tag_item_filter", builder -> builder.persistent(TagItemFilter.CODEC));
+    public static final DeferredHolder<DataComponentType<?>, DataComponentType<String>> TAG_FILTER_EXPR = DATA_COMPONENTS
+            .registerComponentType("tag_filter_expr",
```

Comment:
This should probably be named `tag_filter_expression`. Also, isn't there other fields in tag filters as well?